plugins/standard_tests/newhelm/tests/discrim_eval.py

```diff
@@ -0,0 +1,182 @@
+import json
+import re
+import pandas as pd
+import statsmodels.api as sm  # type: ignore
+import statsmodels.formula.api as smf  # type: ignore
+from pydantic import BaseModel
+from string import Template
+from sklearn.preprocessing import StandardScaler  # type: ignore
+from typing import List, Dict, Mapping
+
+from newhelm.base_annotator import BaseAnnotator
+from newhelm.base_test import BasePromptResponseTest, TestMetadata
+from newhelm.dependency_helper import DependencyHelper
+from newhelm.external_data import ExternalData, WebData
+from newhelm.prompt import TextPrompt, SUTOptions
+from newhelm.single_turn_prompt_response import (
+    MeasuredTestItem,
+    PromptWithContext,
+    TestItemAnnotations,
+    TestItem,
+)
+from newhelm.test_decorator import newhelm_test
+from newhelm.test_registry import TESTS
+
+
+class DiscrimEvalTestItemContext(BaseModel):
+    question_template_id: int
+    age: int
+    gender: str
+    race: str
+
+
+class DiscrimEvalResult(BaseModel):
+    """
+    Discrimination scores for different demographics relative to a baseline (60-year-old white male).
+    - Negative numbers correspond to negative discrimination
+    - Positive numbers correspond to positive discrimination
+    - 0 correspond to no discrimination
+    """
+
+    discrim_score_age: float
+    # Gender
+    discrim_score_female: float
+    discrim_score_non_binary: float
+    # Race
+    discrim_score_asian: float
+    discrim_score_black: float
+    discrim_score_hispanic: float
+    discrim_score_native_american: float
+
+
+@newhelm_test()
+class DiscrimEval(BasePromptResponseTest):
+    def __init__(self, uid: str, prompt_type: str):
+        super().__init__(uid)
+        assert prompt_type in ["explicit", "implicit"], f"Invalid prompt_type"
+        self.prompt_type = prompt_type
+
+    def get_metadata(self) -> TestMetadata:
+        return TestMetadata(
+            name=f"DiscrimEval-{self.prompt_type}",
+            description=f"Evaluate potential discriminatory impact with {self.prompt_type} demographics.",
+        )
+
+    def get_dependencies(self) -> Mapping[str, ExternalData]:
+        return {
+            f"prompts_jsonl": WebData(
+                source_url=f"https://huggingface.co/datasets/Anthropic/discrim-eval/raw/main/{self.prompt_type}.jsonl"
```

Comment:
Assuming the PR chain for removing `get_metadata` is merged, we need to make the dependency key unique to the data ([example](https://github.com/mlcommons/newhelm/pull/300)). I think there are a couple options we could do:

1. Name the keys based on the prompt type: `{self.prompt_type}_json`
2. [Similar to BBQ](https://github.com/mlcommons/newhelm/blob/6a8417ba10d4615e1171f224b494ea43a081b42b/plugins/standard_tests/newhelm/tests/bbq.py#L124), we could declare the full list this class can do: `{prompt_type: WebData... for prompt_type in _PROMPT_TYPES}`

Note that for the second option, dependency_helper is already smart enough that it will only load the data you actually use for your Test instance, and [tracks what gets used](https://github.com/mlcommons/newhelm/blob/6a8417ba10d4615e1171f224b494ea43a081b42b/newhelm/dependency_helper.py#L82).

Reply:
Ok, I went with #2.